htdocs/core/class/commonobject.class.php

```diff
@@ -1262,6 +1262,20 @@ public function add_contact($fk_socpeople, $type_contact, $source = 'external',
 			return -2;
 		}
 
+		if ($this->restrictiononfksoc && ! $user->hasRight('societe', 'client', 'voir')) {
+			$allowed_thirdparties = $user->getAffectedThirdparties(1);
+			$sql_allowed_contacts = 'SELECT rowid FROM '.$this->db->prefix().'socpeople';
+			$sql_allowed_contacts.= ' WHERE fk_soc IN ('.$this->db->sanitize(implode(',', $allowed_thirdparties)).')';
```

Comment:
The number of element into a IN has a limit (generally errors appears when we are over max_allowed_packet) into size of the subrequest.

Instead of dowing a getAffectedThirdparties to get all thirdparties allowed by a user and then a select to get all socpeople for thes thirdparies,  why not just doing a 

SELECT COUNT(*) FROM llx_societe as s JOIN llx_socpeople as sp...

Reply:
You are right, I did it this way.